File: BankAccount.java
public class BankAccount {

    private long accountNumber;
    private double amount;
    private char type;

    public BankAccount(long accountNumber, double amount, char type) {
        this.accountNumber = accountNumber;
        this.amount = amount;
        this.type = type;
    }

    public long getAccountNumber() {
        return accountNumber;
    }

    public void setAccountNumber(long accountNumber) {
        this.accountNumber = accountNumber;
    }

    public double getAmount() {
        return amount;
    }

    public void addMoney(double amount) {
        if (type == 'A' && amount < 50000){
            this.amount = amount;
        } else if (type == 'B' && amount <100000){
            this.amount = amount;
        } else {
            this.amount = amount;
        }
    }

    public void withdraw(double wd){
        amount -= wd;
    }

    public char getType() {
        return type;
    }

    public void setType(char type) {
        if (type == 'A' || type == 'B' || type == 'C') {
            this.type = type;
        } else {
            System.out.println("El tipo de cuenta no existe");
        }
    }

}

File: bankAccount.py
class BankAccount:

    def getType(self):
        return self.__type
    
    def getAccountNumber(self):
        return self.__accountNumber
    
    def getAmount(self):
        return self.__amount

    def __init__(self, accountNumber, type, amount):
        self.__type = type
        self.__accountNumber = accountNumber
        self.__amount = amount
        

    def withdraw(self, wd):
        if wd > self.__amount:
            print("Retiro mayor al saldo de la cuenta")
        elif self.__type == 'A' and wd >= 1000 : 
            self.__amount -= wd
        elif self.__type == 'B' and wd >= 5000:   
            self.__amount -= wd  
        elif self.__type == 'C' and wd >= 10000:   
            self.__amount -= wd
        else: 
            print("Retiro menor al minimo de la cuenta")                                       